core/src/genesis.rs

```diff
@@ -147,55 +147,44 @@ impl GenesisNetworkTrait for GenesisNetwork {
         genesis_config: Option<&Configuration>,
         tx_limits: &TransactionLimits,
     ) -> Result<Option<GenesisNetwork>> {
-        #![allow(clippy::unwrap_in_result)]
-        #![allow(clippy::expect_used)]
         if !submit_genesis {
             iroha_logger::debug!("Not submitting genesis");
             return Ok(None);
         }
         iroha_logger::debug!("Submitting genesis.");
-        Ok(Some(GenesisNetwork {
-            transactions: raw_block
-                .transactions
-                .iter()
-                .map(|raw_transaction| {
-                    let genesis_key_pair = KeyPair::new(
-                        genesis_config
-                            .as_ref()
-                            .expect("Should be `Some` when `submit_genesis` is true")
-                            .account_public_key
-                            .clone(),
-                        genesis_config
-                            .as_ref()
-                            .expect("Should be `Some` when `submit_genesis` is true")
-                            .account_private_key
-                            .clone()
-                            .ok_or_else(|| eyre!("Genesis account private key is empty."))?,
-                    )?;
-
-                    raw_transaction.sign_and_accept(genesis_key_pair, tx_limits)
-                })
-                .enumerate()
-                .filter_map(|(i, res)| {
-                    res.map_err(|error| {
-                        let error_msg = format!("{error:#}");
-                        iroha_logger::error!(error = %error_msg, "Genesis transaction #{i} failed")
-                    })
-                    .ok()
+        let genesis_config =
+            genesis_config.expect("Should be `Some` when `submit_genesis` is true");
+        let genesis_key_pair = KeyPair::new(
+            genesis_config.account_public_key.clone(),
+            genesis_config
+                .account_private_key
+                .clone()
+                .ok_or_else(|| eyre!("Genesis account private key is empty."))?,
+        )?;
+        let transactions = raw_block
+            .transactions
+            .iter()
+            .map(|raw_transaction| {
+                raw_transaction.sign_and_accept(genesis_key_pair.clone(), tx_limits)
+            })
+            .enumerate()
+            .filter_map(|(i, res)| {
+                res.map_err(|error| {
+                    let error_msg = format!("{error:#}");
+                    iroha_logger::error!(error = %error_msg, "Genesis transaction #{i} failed")
```

Comment:
```suggestion
                    iroha_logger::error!(error = %error_msg, transaction_num=i, "Genesis transaction failed")
```
Structured logging will let you extract the value instead of parsing the message.